### ISSUE TITLE
NumberInput Web Component Accessibility Review

### ISSUE THREAD

**Issue by pdavies88** [department-of-veterans-affairs/va.gov-team#39674]
## Description
We need to conduct a full accessibility review of the new NumberInput web component. The component can be viewed in Storybook here: https://60f9b557105290003b387cd5-rucqzrsror.chromatic.com/?path=/docs/components-va-number-input--default

## Acceptance Criteria
- [ ] Full accessibility review of the NumberInput component is completed
- [ ] Add descriptions of any issues found to this ticket
    - [ ] If any problems are found, add screen recordings and/or screenshots to comments on this ticket that show the problem.
- [ ] Alert Katy and the developer who built the component when the review is complete
- [ ] Re-review any fixes the developer makes to the component

**Comment by CourtneyChambers77** [comment]
I have test this component and it does pass

**Comment by pdavies88** [comment]
@CourtneyChambers77 @SarahKay8 looking for some guidance on the best approach to use `aria-describedby` with an `<input>` element. Following the documentation set forth [here](https://www.tpgi.com/describing-aria-describedby/) to me it reads that whatever value that is set for `aria-describedby` should be linked to another HTML element that has the same corresponding value set as an `id` attribute. The corresponding element should then provide a user with additional context about what the `<input>` field would be looking for a user to include.

If this interpretation is correct I believe that we would need to update our `va-number-input` and `va-text-input` components. Because right now we have a single aria-describedby prop that gets added to an `<input>` element but there is no corresponding second element that has that prop as an `id` and a message about what should be entered into the `<input>` element. 

Therefore I think we need to split the current `ariaDescribedby` prop and split it into two props one for `ariaDescribedbyId` and `ariaDescribedbyMessage`. The id prop would give us the linking value between the `<input>` and other element, while the message prop would give us the context of what is included with the input element.

Additionally should these be set by default meaning they should always exist on the component? Right now we have them as an optional prop so a developer would need to build it in to their component for it to exist.

If we do split the prop into two I think that ultimately our code for these input components should look something like:
      <Host>
        {this.label && (
          <label htmlFor="inputField" part="label">
            {this.label}{' '}
            {this.required && <span class="required">(*Required)</span>}
          </label>
        )}
        {this.error && (
          <span class="error-message" role="alert">
            <span class="sr-only">Error:</span> {this.error}
          </span>
        )}
        <input
          id="inputField"
          type="number"
          aria-describedby={
            this.ariaDescribedbyId ? this.ariaDescribedbyId : null
          }
          inputmode={this.inputmode ? this.inputmode : null}
          name={this.name}
          max={this.max}
          min={this.min}
          value={this.value}
          onInput={this.handleInput}
          onBlur={this.handleBlur}
          part="input"
        />
        {this.ariaDescribedbyMessage && (
          <span
            id={this.ariaDescribedbyId ? this.ariaDescribedbyId : null}
            class="sr-only"
          >
            {this.ariaDescribedbyMessage}
          </span>
        )}
      </Host>

Please let me know if this is correct and I will add a ticket to do that work.
@bkjohnson @k80bowman

**Comment by SarahKay8** [comment]
hey @pdavies88 

Sorry on late reply was wrapped up in domo work. 


<label for=”theInput” … >This gets read when the input is in focus. No ARIA needed.</label>

<input id=”theInput” … >

Aside from that you are good to go

**Comment by SarahKay8** [comment]
@pdavies88 
 for example, you have a paragraph <p> that describes the input in more detail, then you can connect that <p id=”description”> with the input using:

 <input … aria-describedby=”description”> but usually extra descriptions aren’t needed if your <label> does its job

**Comment by SarahKay8** [comment]
for example, you have a paragraph < p > that describes the input in more detail, then you can connect that < p id=" description " > with the input using : <input ... aria-describedby="description"> but usually extra descriptions aren't needed if your < label > does its job :)

**Comment by SarahKay8** [comment]
@shiragoodman
